src/storage/DataAccessorBasedStore.ts

```diff
@@ -196,12 +196,10 @@ export class DataAccessorBasedStore implements ResourceStore {
       throw new ConflictHttpError(`${identifier.path} conflicts with existing path ${oldMetadata.identifier.value}`);
     }
 
-    const isContainer = this.isNewContainer(representation.metadata, identifier.path);
+    const isContainer = this.isNewContainer(representation.metadata);
     // Solid, §3.1: "Paths ending with a slash denote a container resource."
     // https://solid.github.io/specification/protocol#uri-slash-semantics
-    if (isContainer !== isContainerIdentifier(identifier)) {
-      throw new BadRequestHttpError('Containers should have a `/` at the end of their path, resources should not.');
-    }
+    this.validateSlug(isContainer, identifier.path);
```

Comment:
This is the cause of your tests failing. When doing a PUT it is not required to have the correct link headers

> When creating new resources, servers can determine an effective request URI’s type by examining the URI path ending

So adding checks here is going to cause all calls that depend on that behaviour to fail.

Reply:
I got them all checked now. Changed the name of `isNewContainer()` to `isContainerType` to better accommodate what it does now. (only checking the metadata)